dev/shiza16/modules.py

```diff
@@ -0,0 +1,256 @@
+    
+    """ Confusion Matrix """
+           
+    plt.figure(figsize = (15,5))
+    plt.xlabel('Actual Vehicle Labels')
+    plt.ylabel('Predicted Vehicle Labels')
+    cmatrix = confusion_matrix(y_test , y_predict , labels = label)
+    matrix = pd.DataFrame(cmatrix, index=label, columns=label)
+    sns.heatmap(matrix, annot=True, linewidths=.5)
+    plt.show()
+
+def model_classification_report(y_test, y_predict):
+    """  Model Classification report for Precision , Recall and F1-Score """
+    print("\n DataSet Report: ")
+    print(classification_report(y_test, y_predict))
+    
+    
+=======
+from sklearn.linear_model import LogisticRegression
+from sklearn.metrics import classification_report, confusion_matrix
+from sklearn.model_selection import GridSearchCV
+from sklearn.metrics import make_scorer
+from sklearn.model_selection import learning_curve
+from sklearn.metrics import f1_score
+from sklearn import metrics
+from sklearn.model_selection import cross_val_score
+
+
+def dataset_statistics(data):
+
+    """ Data Features and Labels"""
+
+    print("Shape of the dataset: ", data.shape)
+
+    print("\nFeatures of the dataset are: ", list(data.keys()))
+
+    print("\nTarget Labels of dataset are: ", data["Class"].unique())
+
+
+def data_visualization(dataa):
+
+    """ For Visulaization of DataSet"""
+
+    print("\nHistogram for analyzing the frequency of labled class.\n")
+    base_color = sns.color_palette()[9]
+    Vorder = dataa["Class"].value_counts().index
+    sns.countplot(data=dataa, x="Class", color=base_color, order=Vorder)
+
+    print("\n")
+    plt.figure(figsize=(25, 15))
+    sns.heatmap(dataa.corr(), annot=True, linewidths=0.5)
+    plt.show()
+    print("Correlation Analysis.")
+
+
+def splitting_train_test_data(data):
+
+    """ Data is splitted into 30:70 for training and testing"""
+    X = data.drop(["Class"], axis=1)
+    y = data["Class"]
+
+    return train_test_split(X, y, test_size=0.3, random_state=45)
+
+
+def SVM_train(X, y):
+
+    """ SVM Classifier"""
+    # kernel =' poly ' is taking infinte time that's why it is not added.
+    params_grid = [{"kernel": ["rbf"], "gamma": [1e-3, 1e-4], "C": [1, 10, 100, 1000]}]
+
+    svm_grid = GridSearchCV(SVC(), params_grid, cv=5)
+    svm_grid = svm_grid.fit(X, y)
+    classifier = SVC(
+        kernel=svm_grid.best_estimator_.kernel,
+        C=svm_grid.best_estimator_.C,
+        gamma=svm_grid.best_estimator_.gamma,
+    )
+
+    return classifier.fit(X, y)
+
+
+def LogisticRegression_train(X, y):
+
+    """ SVM Classifier"""
+    classifier = LogisticRegression(solver="lbfgs", multi_class="auto", max_iter=7600)
+    return classifier.fit(X, y)
+
+
+def Tuning_LogiticRegression(X, y, x):
+    """ Tuning Logistic Regression to increase the accuracy of model """
+
+    parameters = {
+        "penalty": ["l1", "l2"],
+        "C": [0.0001, 0.001, 0.01, 0.1, 1, 10, 100, 1000],
+        "solver": ["liblinear", "saga"],
+        "class_weight": [
```

Comment:
Did you mean to include this? I don't think this does what you want because you have 4 classes, so this will exclude 2 of them IIUC.